File: iPad/VerBank-CSTSv3-IOSClientApi/ClientStation/api/info/InfoOperator_TRADESERV1012.m
//
//  InfoOperator_TRADESERV1012.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "InfoOperator_TRADESERV1012.h"

#import "Info_TRADESERV1012.h"
#import "APIDoc.h"
#import "API_IDEventCaptain.h"

@implementation InfoOperator_TRADESERV1012

- (void)onInfo:(InfoFather *)info {
    Info_TRADESERV1012 *tinfo = (Info_TRADESERV1012 *)info;
    if ([[APIDoc getUserDocCaptain] getCDS_AccountStore:[tinfo getAccountID]] == nil) {
        return;
    }
    
    if ([tinfo getToRemoveOrderID] > 0) {
        long long orderID = [tinfo getToRemoveOrderID];
        TOrder *tempOrder = [[APIDoc getUserDocCaptain] getOrder:orderID];
        if ([tempOrder getCorrespondingTicket] > 0) {
            TTrade *tempTrade = [[APIDoc getUserDocCaptain] getTrade:[tempOrder getCorrespondingTicket]];
            if (tempTrade != nil) {
                [tempTrade setCorOrderID:0];
            }
        }
        [[APIDoc getUserDocCaptain] removeOrder:orderID];
    }
    
    if ([tinfo getOrder] != nil) {
        [[APIDoc getUserDocCaptain] addOrder:[tinfo getOrder]];
        if ([[tinfo getOrder] getCorrespondingTicket] > 0) {
            TTrade *tempTrade = [[APIDoc getUserDocCaptain] getTrade:[[tinfo getOrder] getCorrespondingTicket]];
            if (tempTrade != nil) {
                [tempTrade setCorOrderID:[[tinfo getOrder] getOrderID]];
            }
        }
    }
    
    [API_IDEventCaptain fireEventChanged:DATA_ON_Order_Changed eventData:[@([tinfo getAccountID]) stringValue]];
    if ([[tinfo getOrder] getCorrespondingTicket] > 0) {
        [API_IDEventCaptain fireEventChanged:DATA_ON_Trade_Changed eventData:[@([tinfo getAccountID]) stringValue]];
    }
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/info/InfoOperator_TRADESERV1012.java
package jedi.ex01.client.station.api.info;

import jedi.ex01.CSTS3.comm.info.InfoFather;
import jedi.ex01.CSTS3.comm.info.Info_TRADESERV1012;
import jedi.ex01.CSTS3.comm.struct.TOrders4CFD;
import jedi.ex01.CSTS3.comm.struct.TTrade4CFD;
import jedi.ex01.client.station.api.doc.DataDoc;
import jedi.ex01.client.station.api.event.API_IDEventCaptain;
import jedi.ex01.client.station.api.event.API_IDEvent_NameInterface;

public class InfoOperator_TRADESERV1012 extends InfoOperator {

	public void onInfo(InfoFather _info) {
		Info_TRADESERV1012 info = (Info_TRADESERV1012) _info;
		if (DataDoc.getInstance().getAccountStore().getAccountID() != info.getAccountID()) {
			return;
		}
		if (info.getToRemoveOrderID() > 0) {
			Long orderID = new Long(info.getToRemoveOrderID());
			TOrders4CFD temporder = DataDoc.getInstance().getOrder(orderID);
			if (temporder.getCorrespondingTicket() > 0) {
				TTrade4CFD temptrade = DataDoc.getInstance().getTrade(temporder.getCorrespondingTicket());
				if (temptrade != null) {
					temptrade.setCorOrderID(0);
				}
			}
			DataDoc.getInstance().removeOrder(orderID);
		}
		if (info.getOrder4CFD() != null) {
			DataDoc.getInstance().addOrder(info.getOrder4CFD());
			if (info.getOrder4CFD().getCorrespondingTicket() > 0) {
				TTrade4CFD temptrade = DataDoc.getInstance().getTrade(
						info.getOrder4CFD().getCorrespondingTicket());
				if (temptrade != null) {
					temptrade.setCorOrderID(info.getOrder4CFD().getOrderID());
				}
			}
		}
		API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.DATA_ON_Order_Changed);
		if (info.getOrder4CFD().getCorrespondingTicket() > 0) {
			API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.DATA_ON_Trade_Changed);
		}
	}

}
